spec/federation_messages/federation_messages_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require "federation-testbed"
+
+describe "Generation and dispatch of federation messages" do
```

Comment:
These are ultimately integration tests, so how about a subfolder in `spec/integration`?